app/models/in_progress_form.rb

```diff
@@ -1,8 +1,16 @@
 # frozen_string_literal: true
 class InProgressForm < ActiveRecord::Base
   attr_encrypted :form_data, key: Settings.db_encryption_key
+  validates(:form_data, presence: true)
+  before_save :serialize_form_data
 
   def self.form_for_user(form_id, user)
     InProgressForm.find_by(form_id: form_id, user_uuid: user.uuid)
   end
+
+  private
+
+  def serialize_form_data
+    self.form_data = form_data.to_json unless form_data.is_a?(String)
```

Comment:
we should keep most of the processing to the backend though because on the frontend some people have really slow phones, on the backend we have fast servers and can always spin up more if needed.

this amount of conversions looks like the only solution for dealing with having 2 fields with one encrypted and the other not. if we only had one field we could do read_attribute on that field and send that over skipping all conversions for the GET request.